### PR TITLE
[Enhancement](be-logger) Support custom date time format functionality in be log.

### PR DESCRIPTION
**Pull request by kaka11chen**
## Proposed changes

Add these configurations to control it.

```
// log enable custom date time format
DEFINE_Bool(sys_log_enable_custom_date_time_format, "false");
// log custom date time format (https://en.cppreference.com/w/cpp/io/manip/put_time)
DEFINE_String(sys_log_custom_date_time_format, "%Y-%m-%d %H:%M:%S");
// log custom date time milliseconds format (fmt::format)
DEFINE_String(sys_log_custom_date_time_ms_format, ",{:03d}");
```

### CODE REVIEW

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

**Comment by hello-stephen** [comment]
run compile

File: be/src/common/logconfig.cpp [hunk 1]
```diff
@@ -49,21 +49,41 @@ static bool iequals(const std::string& a, const std::string& b) {
...
-    s << std::setw(2) << l.time.day();
-    s << ' ';
-    s << std::setw(2) << l.time.hour() << ':';
-    s << std::setw(2) << l.time.min() << ':';
-    s << std::setw(2) << l.time.sec() << ".";
-    s << std::setw(6) << l.time.usec();
+
+    // Add a space if custom_date_time_format.
+    if constexpr (custom_date_time_format) {
+        s << ' ';
```
Comment (morningman) [inline be/src/common/logconfig.cpp:65]: Why need a space here?

File: be/src/common/logconfig.cpp [hunk 2]
```diff
@@ -49,21 +49,41 @@ static bool iequals(const std::string& a, const std::string& b) {
...
+    tm_time.tm_mday = l.time.day();
+    tm_time.tm_hour = l.time.hour();
+    tm_time.tm_min = l.time.min();
+    tm_time.tm_sec = l.time.sec();
+
+    if constexpr (custom_date_time_format) {
+        s << std::put_time(&tm_time, config::sys_log_custom_date_time_format.c_str());
+        if (!config::sys_log_custom_date_time_ms_format.empty()) {
+            s << fmt::format(config::sys_log_custom_date_time_ms_format,
+                             l.time.usec() / 1000); // 确保毫秒为三位数字
```
Comment (morningman) [inline be/src/common/logconfig.cpp:80]: English comment

**Comment by kaka11chen** [comment]
run buildall

**Comment by kaka11chen** [comment]
run buildall

File: be/src/common/logconfig.cpp (hunk 1 above)
Comment (kaka11chen) [reply be/src/common/logconfig.cpp:65]: easy to parse date time

**Comment by kaka11chen** [comment]
run buildall

**Comment by hello-stephen** [comment]
run compile

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

**Pull request merged**